.github/workflows/coverage.yml

```
@@ -0,0 +1,209 @@
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-report-xml
+          path: coverage.xml
+          retention-days: 30
+
+      - name: Generate diff coverage report (PR only)
+        if: github.event_name == 'pull_request'
+        run: |
+          # Generate diff coverage report
+          uv run diff-cover coverage.xml --compare-branch=origin/main --html-report diff-coverage.html --markdown-report diff-coverage.md
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)